### PR TITLE
Added support for create/delete file shares.

### PR DESCRIPTION
**Pull request by katmsft**

### CODE REVIEW

File: sdk/storage/inc/shares/protocol/share_rest_client.hpp [hunk 1]
```diff
@@ -2158,9 +2088,12 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
           request.AddQueryParameter(
               Details::c_QueryTimeout, std::to_string(createOptions.Timeout.GetValue()));
         }
-        for (const auto& pair : createOptions.Metadata.GetValue())
+        if (createOptions.Metadata.HasValue())
```
Comment (JinmingHu-MSFT) [inline sdk/storage/inc/shares/protocol/share_rest_client.hpp:2091]: wait a minute, we already made a conclusion that metadata isn't optional. What're you doing here?

File: sdk/storage/inc/shares/protocol/share_rest_client.hpp [hunk 2]
```diff
@@ -2258,9 +2191,8 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
             Timeout; // The timeout parameter is expressed in seconds. For more information, see <a
                      // href="https://docs.microsoft.com/en-us/rest/api/storageservices/Setting-Timeouts-for-File-Service-Operations?redirectedfrom=MSDN">Setting
                      // Timeouts for File Service Operations.</a>
-        Azure::Core::Nullable<std::map<std::string, std::string>> Metadata
-            = std::map<std::string, std::string>(); // A name-value pair to associate with a file
-                                                    // storage object.
+        Azure::Core::Nullable<std::map<std::string, std::string>>
```
Comment (JinmingHu-MSFT) [inline sdk/storage/inc/shares/protocol/share_rest_client.hpp:2194]: This is wrong

File: sdk/storage/inc/shares/share_client.hpp [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+     * @brief Creates the file share.
+     * @param options Optional parameters to create this file share.
+     * @return Azure::Core::Response<ShareInfo> The information containing the version and modified
+     * time of a share.
+     */
+    Azure::Core::Response<ShareInfo> Create(
+        const CreateShareOptions& options = CreateShareOptions()) const;
+
+    /**
+     * @brief Deletes the file system.
```
Comment (JinmingHu-MSFT) [inline sdk/storage/inc/shares/share_client.hpp:83]: not filesystem

File: sdk/storage/inc/shares/protocol/share_rest_client.hpp (hunk 1 above)
Comment (katmsft) [reply sdk/storage/inc/shares/protocol/share_rest_client.hpp:2091]: Oops, forgot it, sorry.

File: sdk/storage/inc/shares/share_client.hpp [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+     * @param options Optional parameters to delete this file system.
```
Comment (JinmingHu-MSFT) [inline sdk/storage/inc/shares/share_client.hpp:84]: not file system, ditto for others

File: sdk/storage/inc/shares/share_options.hpp [hunk 5]
```diff
@@ -57,4 +66,41 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
...
+  {
+    /**
+     * @brief Context for cancelling long running operations.
+     */
+    Azure::Core::Context Context;
+
+    /**
+     * @brief A name-value pair to associate with a file storage object.
+     */
+    Azure::Core::Nullable<std::map<std::string, std::string>> Metadata;
```
Comment (JinmingHu-MSFT) [inline sdk/storage/inc/shares/share_options.hpp:79]: we've made a conclusion that `Metadata` isn't optional.

**Pull request merged**